### PR TITLE
Shiny weather can2.0 wip

### PR DESCRIPTION
**Pull request by nickyrong**
- Interactivity/data flow reworked (station ID is now a selection list like ShinyCFA)
- Spinner uses waiter{} package
- % missing data plot upgraded
- Successfully hosted on shinyapps.io (https://nickrongkp.shinyapps.io/WeatherCan/)

*Be aware of memory usage!* When testing on the deployed app on shinyapps.io, app crashes due to "out of memory" when trying to plot % missing on daily data with a station that has >100 year of data.

### CODE REVIEW

File: SS/server_old.R [hunk 1]
```diff
@@ -0,0 +1,320 @@
+rm(list = ls())
```
Comment (WraySmith) [inline SS/server_old.R:1]: if this is going to be moved into an SS folder, just move the old file without a rename - everything is tracked via commits anyways

File: SS/ui_old.R [hunk 2]
```diff
@@ -0,0 +1,81 @@
+rm(list=ls())
```
Comment (WraySmith) [inline SS/ui_old.R:1]: same comment for this file with regards to rename and move

File: app/server.R [hunk 3]
```diff
@@ -1,15 +1,8 @@
...
-# For servers
-#list of packages required
-#list.of.packages <- c("tidyverse", "weathercan", "lutz", "sf", "DT", "naniar", "plotly")
-#checking missing packages from list
-#new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
-#install missing ones
-#if(length(new.packages)) install.packages(new.packages, dependencies = TRUE)
-
-library(tidyverse) # we live in the tidyverse!
+library(tidyr) # we live in the tidyverse!
```
Comment (WraySmith) [inline app/server.R:3]: remove random statements, just put in what it's used for

File: app/server.R [hunk 4]
```diff
@@ -20,301 +13,487 @@ library(plotly) # interactive available data plotting
...
-                            base::as.Date()
     
-    output$info_date <- renderText({current.csv.date %>% as.character()})
+  })
+  
+  # Sometimes monthly interval has shorter record length
+  output$stn_warning <- renderText({
     
-    # Update meta-data list if there's no file or the file is older than 7 days
-    if(is.na(current.csv.date) || (Sys.Date() - current.csv.date) > 7) {
```
Comment (WraySmith) [inline app/server.R:66]: is this not being done? someone is going to click the update button for sure everytime they're on the page...

**Review by WraySmith — APPROVED** [review]
In general, the way the files were modified make it pretty hard to review the dif. Instead of writing blocks of code in a separate file and then pasting it in place, it's easier to compare if it's edited in place so the differences show up generally where the differences exist. As such, in terms of comparing differences I didn't really review. I looked at the app and it's looking good. 

So I think just merge this (after resolving a few of the comments above) and I'll create a branch and make edits.

**Comment by nickyrong** [comment]
Comments make sense. I wasn't thinking of Pull Request when I started working on revising the apps (you can tell by how I renamed the old files xx_old.R and have them in /SS). Still learning how to do production on git.

**Pull request merged**